tmt/utils.py

```diff
@@ -1318,6 +1318,58 @@ def items(self) -> Generator[Tuple[str, Any], None, None]:
 
         yield from self.to_dict().items()
 
+    @classmethod
+    def default(cls, key: str, default: Any = None) -> Any:
+        """
+        Return a default value for a given key.
+
+        Keys may have a default value, or a default *factory* has been specified.
+
+        :param key: key to look for.
+        :param default: when key has no default value, ``default`` is returned.
+        :returns: a default value defined for the key, or its ``default_factory``'s
+            return value of ``default_factory``, or ``default`` when key has no
+            default value.
+        """
+
+        for field in dataclasses.fields(cls):
+            if key != field.name:
+                continue
+
+            if not isinstance(field.default_factory, dataclasses._MISSING_TYPE):
+                return field.default_factory()
+
+            if not isinstance(field.default, dataclasses._MISSING_TYPE):
+                return field.default
+
+        else:
+            return default
+
+    @property
+    def is_bare(self) -> bool:
```

Comment:
@happz I am thinking about the naming here, would not something like `is_unchanged` be more descriptive?

Reply:
I was looking for a name to express that the instance has not been touched or modified. `unchanged` sounds more like a "has it changed since the last time I looked", so not really fit my idea, but I'm open to suggestions.

Reply:
yeah, dunno, naming is hard. I can live with `is_bare`